Tests/Unit/NetworkServer/LoRaDeviceRegistryTest.cs

```diff
@@ -44,6 +44,28 @@ public async Task GetDeviceForJoinRequestAsync_When_Device_Api_Throws_Error_Shou
             apiService.VerifyAll();
         }
 
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task GetDeviceForJoinRequestAsync_When_Join_Handled_By_Other_Cache_Is_Updated(bool joinedDevice)
+        {
+            var devNonce = new DevNonce(1);
+            var apiService = new Mock<LoRaDeviceAPIServiceBase>();
+            var otaaDevice = TestDeviceInfo.CreateOTAADevice(1);
+            if (joinedDevice) otaaDevice.AppSKey = "123";
```

Comment:
[nit] you could create the key here using `TestKeys` (will make the build green again)

Reply:
Already fixed. Thanks. Didn't know about the TestKeys. Will use them when I need real keys.